tests/test_templates.py

```diff
@@ -0,0 +1,752 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Integration tests for the ``execute`` method of Lightning devices.
+"""
+import functools
+
+import pennylane as qml
+import pytest
+from conftest import LightningDevice, device_name
+from pennylane import numpy as np
+
+# pylint: disable=missing-function-docstring, too-few-public-methods
+
+if LightningDevice._new_API and not LightningDevice._CPP_BINARY_AVAILABLE:
```

Comment:
I'm not sure if this condition for skipping is right. Could you please double-check?

Reply:
| newAPI    | not binary | skip |
| -------- | ------- | ------- |
| True  | True    | True |
| True | False    | False |
| False    | True    | False |
| False    | False    | False |

right?

Reply:
Right.

Reply:
We could always skip if `not LightningDevice._CPP_BINARY_AVAILABLE` is that what you're thinking?

Reply:
Yes. This looks like more related to having binaries than the device API unless something is escaping me here.